wolfcrypt/src/hmac.c

```diff
@@ -46,7 +46,8 @@
     {
         if (hmac == NULL || (key == NULL && keySz != 0) ||
            !(type == WC_MD5 || type == WC_SHA || type == WC_SHA256 ||
-                type == WC_SHA384 || type == WC_SHA512 || type == BLAKE2B_ID)) {
+                type == WC_SHA384 || type == WC_SHA512 ||
```

Comment:
My reading of this code is that you can't set the HMAC key with SHA-3 algorithm type in FIPS.
Is this correct? Are we validating HMAC-SHA-3?

Reply:
We are not adding SHA-3 to the existing FIPS boundary. We are adding SHA-3 and HMAC-SHA-3 to the new FIPS boundary for that is a WIP. This PR is for non-FIPS HMAC-SHA-3.